src/components/pages/Notes/NoteGroupNav.js

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import { connect } from "react-redux";
+import { values } from "lodash";
+import { NavLink, Route } from 'react-router-dom';
+
+import Button from "../../library/Button";
+
+
+function NoteGroupNav({ children, noteGroups, addGroup, deleteGroup, updateGroupName }) {
+  const [editedGroupId, setEditedGroupId] = useState();
+  const [editedGroupName, setEditedGroupName] = useState('jeanPaulBelmondo');
+
+  return (
+    <div className="group-nav">
+      <div className="columns">
+        <div className="column is-2">
+          <aside className="menu">
+            <ul className="menu-list">
+              {noteGroups.map(noteGroup => {
+                return(
+                    <li key={noteGroup.id}>
+                      {editedGroupId === noteGroup.id ? (
+                        <input
+                          className="input is-primary is-small"
+                          autoFocus
+                          value={editedGroupName}
+                          onChange={e => setEditedGroupName(e.target.value)}
```

Comment:
You want to catch to specific keyboard events when the user is typing in the input: `Enter` and `Escape`.
You can check this website to see what are the keycodes of the events you are interested in : https://keycode.info/.

Then you can pass the property `onKeyDown` to the input. It is a special property provided by React. Each time a keyboard key will be pressed, the function that you provide will be called by React with the corresponding keyboard event.
For example, to see it in action, you can start by passing `onKeyDown={e => console.log(e.keyCode)}`. Then, once you know what keyCodes you want to listen to and what you want to do when the keyCode matches Enter or Escape, you can improve your function.

It would then look something like this:
```jsx
// Before the return of you function, define your onKeyDown function
const onInputKeyDown = e => {
    // Enter
    if (e.keyCode === ???) {
      // updateGroupName
      // reset the editedGroupId to null to stop editing
    }
    
    // Escape
    if (e.keyCode === ???) {
      // reset the editedGroupId to null to stop editing
    }
 }
    ...
    
    <input
        className="input is-primary is-small"
        autoFocus
        value={editedGroupName}
        onChange={e => setEditedGroupName(e.target.value)}
        onKeyDown={onInputKeyDown}
    />
  };
```